sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/package.scala

```diff
@@ -162,7 +162,8 @@ package object util {
   def toCommentSafeString(str: String): String = {
     val len = math.min(str.length, 128)
     val suffix = if (str.length > len) "..." else ""
-    str.substring(0, len).replace("*/", "\\*\\/").replace("\\u", "\\\\u") + suffix
+    str.substring(0, len).replace("*/", "\\*\\/")
+      .replaceAll("(^|[^\\\\])(\\\\(\\\\\\\\)*u)", "$1\\\\$2") + suffix
```

Comment:
Thanks for the suggestion @mhseiden .
I tried escaping by `escapeJava` and it may fix this issue but I noticed it escapes all of "\", means the number of "\" will be doubled.
For example, "\\\u0022" will be "\\\\\\u0022" but I expects only "\" just before "u" will be escaped if the number of "\" is odd.